client/src/pages/GameLobby.tsx

```diff
@@ -1,27 +1,30 @@
 import DisplayLogo from "../components/DisplayLogo";
 
 import { useLoaderData } from "react-router-dom";
+import { useEffect, useState } from "react";
+import { PlayerAttributes } from "../../../types/types.ts";
+import { socket } from "../App.tsx";
 
 const GameLobby = () => {
   const gameData = useLoaderData() as { gameCode: string; qrCode: string };
 
-  // const [mapState, setMapState] = useState(new Map<string, PlayerAttributes>());
-  //
-  // const updateMap = (key: string, value: PlayerAttributes) => {
-  //     setMapState(map => new Map(map.set(key, value)));
-  // }
-  // useEffect(() => {
-  //     socket.on("PLAYER_HAS_JOINED", (player) => {
-  //
-  //         console.log("game trying to be created")
-  //
-  //         if (mapState.has(player.socketId)) {
-  //             // shouldnt happen, but will have to add stuff to prevent it from happening
-  //         } else {
-  //             updateMap(player.socketId, player);
-  //         }
-  //     })
-  // }, [socket]);
+  const [players, setPlayerList] = useState(new Array<PlayerAttributes>());
+
+  const updateList = (player: PlayerAttributes) => {
+    setPlayerList((f) => [...f, player]);
+  };
+
+  useEffect(() => {
+    socket.on("PLAYER_HAS_JOINED", (player) => {
+      console.log(player);
+      updateList(player);
+      console.log(`Player ${player.name} has joined`);
+    });
```

Comment:
Pass an empty dependency array, otherwise this will run on each re-render

Reply:
It is getting passed a dependency array just below